index.js

```diff
@@ -253,18 +284,21 @@ Object.defineProperty(File.prototype, 'extname', {
 
 Object.defineProperty(File.prototype, 'path', {
   get: function() {
-    return this.history[this.history.length - 1];
+    var last = this.history[this.history.length - 1];
+    return this.isDirectory() ? endWithSep(last) : last;
```

Comment:
should `endWithSep` be done inside the setter to make sure paths are stored consistently in the history?  Since people can access history, sometimes `file.history[this.history.length - 1] === file.path` would be false.